src/mme/mme-gn-handler.c

```diff
@@ -69,28 +69,23 @@ static int decode_global_enb_id(S1AP_Global_ENB_ID_t *glob_enb_id, const uint8_t
     return OGS_OK;
 }
 
-/* 3GPP TS 23.003 2.8.2.1 Mapping from GUTI to RAI, P-TMSI and P-TMSI signature */
-void guti_to_rai_ptmsi(const ogs_nas_eps_guti_t *nas_guti, ogs_nas_rai_t *rai, mme_p_tmsi_t *ptmsi, uint32_t *ptmsi_sig)
+/* 3GPP TS 23.003 2.8.2.2 Mapping RAI and P-TMSI from GUTI (in the MME) */
+void guti_to_rai_ptmsi(const ogs_nas_eps_guti_t *nas_guti, ogs_nas_rai_t *rai, mme_p_tmsi_t *ptmsi)
 {
     rai->lai.nas_plmn_id = nas_guti->nas_plmn_id;
-    rai->lai.lac = nas_guti->mme_gid;
-    rai->rac = nas_guti->mme_code;
+    rai->lai.lac = htons(nas_guti->mme_gid);
```

Comment:
Note that I'm not sure why this is needed since I menioned above that the byte order is converted automatically.

Without this the lac would be reported as 256 where it was in fact 1.

Reply:
IIRC I also run across problems with this when implementing the initial patch.
I think it's because nas_guti->mme_gid is in host byte order, but we simply then submit the whole rai as is (as a buffer) in mme_gn_build_sgsn_context_request, so it needs to be converted to netwok byte order.